enumerable.rb

```diff
@@ -0,0 +1,86 @@
+# rubocop:disable Style/For
+
+# Enumerable module
+module Enumerable
+  def my_each
+    return unless block_given?
+
+    for i in self
+      yield i
+    end
+  end
+
+  def my_each_with_index
+    return unless block_given?
+
+    index = 0
+    for i in self
+      yield i, index
+      index += 1
+    end
+  end
+
+  def my_select
+    return unless block_given?
+
+    array = []
+    my_each { |x| array << x if yield x }
+
+    array
+  end
+
+  def my_all?
+    return unless block_given?
+
+    result = my_select { |x| yield x }
+    length == result.length
+  end
```

Comment:
- [ ] It should return false if the block returns false or nil

- [ ] when no block or argument is given, returns true when none of the collection members are false or nil

- [ ] when no block or argument is given returns false when one of the collection members is false or nil

- [ ] when a class is passed as an argument returns true if all of the collection is a member of such class

- [ ] when a class is passed as an argument returns false if any of the collection is not a member of such class

- [ ] when a Regex is passed as an argument returns true if all of the collection matches the Regex

- [ ] when a Regex is passed as an argument returns false if any of the collection does not match the Regex

- [ ] when a pattern other than Regex or a Class is given returns true if all of the collection matches the pattern

- [ ] when a pattern other than Regex or a Class is given returns false if any of the collection does not match the pattern

Reply:
How can we achieve passing a regex as an argument? is this ok?, please help
`def my_all?(*args)
  case args.length 
    when 1 then args.first.is_a?(Regexp) ? reg = args.first : result = args.first
    when 2 then result, reg = args.first, args.last
  end

  result ||= 0

  my_select { |x| result = block_given? ? yield(result, x) : result.send(reg, x) }

  length == result.length
end

p ["ant", "bear", "cat"].my_all?(/t/)`